baseapp/baseapp.go

```diff
@@ -74,8 +79,10 @@ type BaseApp struct { //nolint: maligned
 	//
 	// checkState is set on InitChain and reset on Commit
 	// deliverState is set on InitChain and BeginBlock and set to nil on Commit
-	checkState   *state // for CheckTx
-	deliverState *state // for DeliverTx
+	checkState           *state // for CheckTx
+	deliverState         *state // for DeliverTx
+	processProposalState *state // for PrepareProposal
+	prepareProposalState *state // for ProcessProposal
```

Comment:
Got these backwards :p

```suggestion
	processProposalState *state // for ProcessProposal
	prepareProposalState *state // for PrepareProposal
```